http-clients/src/main/java/org/triplea/http/client/web/socket/GenericWebSocketClient.java

```diff
@@ -49,7 +51,9 @@ public GenericWebSocketClient(
     this.messageListener = messageListener;
     client = webSocketClient;
     client.addListener(this);
-    client.connect();
+    CompletableFutureUtils.logExceptionWhenComplete(
+        CompletableFuture.runAsync(client::connect, threadPool),
```

Comment:
If there is an exception on connect, will that bubble up and notify the user? The underlying code had a parameterized timeout to essentially test that case. Is this failure mode still tested so we can be sure we have the expected behavior?

Reply:
Hmm, good point, I don't think that'll actually be the case. I'll add that